From aa962e654d8e697bc163561aae38d3b785cd0f87 Mon Sep 17 00:00:00 2001
From: Herrie <Github.com@herrie.org>
Date: Fri, 15 Sep 2023 11:18:14 +0200
Subject: [PATCH 5/5] luna-service2: Add permissions for com.palm and
 org.webosports

To solve various issues when trying to register a service from another service such as:

Jun 18 18:15:11 hammerhead webos-telephonyd[1158]: Failed to initialize the Luna Palm service: Invalid permissions for com.palm.telephony
Jun 18 18:15:11 hammerhead webos-telephonyd[1158]: Failed to initialize the WAN service: Invalid permissions for com.palm.wan
Jun 18 18:15:11 hammerhead webos-telephonyd[1158]: Cleaning up

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>
---
 files/sysbus/com.webos.lunasend.role.json.in    | 4 ++++
 files/sysbus/com.webos.lunasendpub.role.json.in | 4 ++++
 files/sysbus/ls-monitor.role.json.in            | 8 ++++++++
 files/sysbus/luna-service2.perm.json            | 6 +++++-
 4 files changed, 21 insertions(+), 1 deletion(-)

diff --git a/files/sysbus/com.webos.lunasend.role.json.in b/files/sysbus/com.webos.lunasend.role.json.in
index 103ba62..e39e000 100644
--- a/files/sysbus/com.webos.lunasend.role.json.in
+++ b/files/sysbus/com.webos.lunasend.role.json.in
@@ -7,6 +7,10 @@
         {
             "service": "com.webos.lunasend-*",
             "outbound": ["*"]
+        },
+        {
+            "service": "org.webosports.lunasend-*",
+            "outbound": ["*"]
         }
     ]
 }
diff --git a/files/sysbus/com.webos.lunasendpub.role.json.in b/files/sysbus/com.webos.lunasendpub.role.json.in
index b90a9c8..6e87372 100644
--- a/files/sysbus/com.webos.lunasendpub.role.json.in
+++ b/files/sysbus/com.webos.lunasendpub.role.json.in
@@ -7,6 +7,10 @@
         {
             "service": "com.webos.lunasendpub-*",
             "outbound": ["*"]
+        },
+        {
+            "service": "org.webosports.lunasend-*",
+            "outbound": ["*"]
         }
     ]
 }
diff --git a/files/sysbus/ls-monitor.role.json.in b/files/sysbus/ls-monitor.role.json.in
index 7b5e894..5245568 100644
--- a/files/sysbus/ls-monitor.role.json.in
+++ b/files/sysbus/ls-monitor.role.json.in
@@ -7,6 +7,14 @@
         {
             "service":"com.webos.monitor*",
             "outbound":["*"]
+        },
+        {
+            "service":"com.palm.monitor*",
+            "outbound":["*"]
+        },
+        {
+            "service":"org.webosports.monitor*",
+            "outbound":["*"]
         }
     ]
 }
diff --git a/files/sysbus/luna-service2.perm.json b/files/sysbus/luna-service2.perm.json
index c976454..d0e24e4 100644
--- a/files/sysbus/luna-service2.perm.json
+++ b/files/sysbus/luna-service2.perm.json
@@ -1,8 +1,12 @@
 {
   "*": ["service.communication", "servicebus.communication"],
   "com.webos.monitor*": ["service.monitor"],
+  "com.palm.monitor*": ["service.monitor"],
+  "org.webosports.monitor*": ["service.monitor"],
   "com.webos.*": ["servicebus.signal"],
   "com.palm.*":  ["servicebus.signal"],
   "com.lge.*":   ["servicebus.signal"],
-  "com.webos.lunasend-*": ["all"]
+  "com.webos.lunasend-*": ["all"],
+  "com.palm.lunasend-*": ["all"],
+  "org.webosports.lunasend-*": ["all"]
 }
-- 
2.41.0.windows.3

